packages/3id-did-resolver/generateVectorDocuments.js

```diff
@@ -77,14 +81,24 @@ const waitForAnchor = doc => new Promise(resolve => {
   })
 })
 
+const makeDID = function(provider, ceramic) {
+  const keyDidResolver = KeyDidResolver.getResolver()
+  const threeIdResolver = ThreeIdResolver.getResolver(ceramic)
+  const resolver = new Resolver({
+    ...threeIdResolver, ...keyDidResolver,
+  })
+  return new DID({ provider, resolver })
+}
+
 const setLegacyDoc = async (ceramic, doc, keyset) => {
   const signing = encodeKey(keyset.signingPub, 'secp256k1')
   const encryption = encodeKey(keyset.encPub, 'x25519')
   const oldProvider = ceramic.did._client.connection
   const provider = new Ed25519Provider(keyset.seed)
-  await ceramic.setDIDProvider(provider)
-  const didstr = ceramic.did.id
-  await ceramic.setDIDProvider(oldProvider)
+  const didWithProvider = makeDID(provider, ceramic)
```

Comment:
The changes in this file were done manually and were not tested.  May be worth testing @oed to make sure I didn't break anything.

I want to state for the record that I don't like the way this `generateVectorDocuments.js` is set up right now.  I feel like if there is any sort of auto-generated files as part of your build or test process, the auto-generation should happen every time as part of the standard build and test process so that it remains current.  A "run once" system where you're supposed to re-run manually at certain special moments (that might not even be clear to everyone when is supposed to happen) seems destined to bitrot and cause headaches down the road.

Reply:
I just included this script for completeness in case something goes very wrong. We should never need to regenerate them, if we need to it means that we broke the http api in a way we are not supposed to break it.

Reply:
What about when we update to V1 of the http api?  It all just seems very brittle to me.  But definitely not high priority to change right now